Update SSH docs

aclevername opened the pull request:
### Description

#3198

### Checklist
- [ ] Added tests that cover your change (if possible)
- [ ] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [ ] Manually tested
- [ ] Made sure the title of the PR is a good description that can go into the release notes
- [ ] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

aclevername commented on userdocs/src/usage/managing-nodegroups.md:
````diff
@@ -160,6 +160,35 @@ There are no specific commands in `eksctl`to update the labels of a nodegroup bu
 kubectl label nodes -l alpha.eksctl.io/nodegroup-name=ng-1 new-label=foo
 ```
 
+### SSH Access
````
We don't really have a great place for these docs to live, and I don't really want to add a page for just SSH configuration. I've put it here for now, open to suggests

michaelbeaumont commented on pkg/apis/eksctl.io/v1alpha5/types.go:
```diff
@@ -1058,20 +1058,22 @@ type (
 		// Enables the ability to [SSH onto nodes using SSM](/introduction#ssh-access)
 		// +optional
-		EnableSSM *bool `json:"enableSsm,omitempty"`
+		EnableSSM *bool `json:"enableSSM,omitempty"`
```
hmm this is a breaking change unfortunately... ugh the `json` tags in go are such a terrible, inflexible abstraction

aclevername replied:
wait really?! I always thought json wasn't case sensitive for key names 😢  I'll change it back...

michaelbeaumont replied:
That'd be nice, check https://github.com/weaveworks/eksctl/issues/753

Callisto13 commented on userdocs/src/usage/managing-nodegroups.md:
````diff
@@ -160,6 +160,35 @@ There are no specific commands in `eksctl`to update the labels of a nodegroup bu
 ```
 
+### SSH Access
+You can enable SSH access for nodegroups by configuring one of `publicKey`, `publicKeyName` and `publicKeyPath` in your
````
```suggestion
You can enable SSH access for nodegroups by configuring one of `publicKey`, `publicKeyName` or `publicKeyPath` in your
```

michaelbeaumont commented on pkg/apis/eksctl.io/v1alpha5/types.go:
```diff
@@ -1058,14 +1058,16 @@ type (
+		// defaults to "~/.ssh/id_rsa.pub", otherwise the value is ignored.
 		PublicKeyPath *string `json:"publicKeyPath,omitempty"`
-		// +optional
+		// +optional Public key to be added to the nodes SSH keychain. If Allow is false this value is ignored.
```
I'm not sure every field needs to have "If Allow is false this value is ignored." It looks pretty noisy at https://deploy-preview-3233--eksctl.netlify.app/usage/schema/#nodeGroups-ssh-allow

michaelbeaumont commented on pkg/apis/eksctl.io/v1alpha5/types.go:
```diff
@@ -1058,14 +1058,16 @@ type (
+		// PublicKeyPath, PublicKey and PublicKeyName can be configured
 		Allow *bool `json:"allow"`
-		// +optional
+		// +optional The path to the SSH public key to be added to the nodes SSH keychain. If Allow is true this value
+		// defaults to "~/.ssh/id_rsa.pub", otherwise the value is ignored.
```
```suggestion
		// The path to the SSH public key to allow access to the nodes
		// Defaults to `"~/.ssh/id_rsa.pub"`
```

michaelbeaumont reviewed (approved):
@aclevername I shortened the strings further (looks better on the schema page), removed the `+optional` tags hanging around which don't do anything and rebased!

aclevername commented:
> @aclevername I shortened the strings further (looks better on the schema page), removed the `+optional` tags hanging around which don't do anything and rebased!

Thanks!
